app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/qhybrid/TimePicker.java

```diff
@@ -0,0 +1,279 @@
+package nodomain.freeyourgadget.gadgetbridge.devices.qhybrid;
+
+import android.app.AlertDialog;
+import android.content.ContentValues;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.pm.PackageInfo;
+import android.graphics.Bitmap;
+import android.graphics.Canvas;
+import android.graphics.Color;
+import android.graphics.Paint;
+import android.util.Log;
+import android.view.MotionEvent;
+import android.view.View;
+import android.view.WindowManager;
+import android.widget.CheckBox;
+import android.widget.CompoundButton;
+import android.widget.ImageView;
+import android.widget.LinearLayout;
+import android.widget.RadioButton;
+import android.widget.RadioGroup;
+import android.widget.ScrollView;
+
+import androidx.annotation.NonNull;
+import nodomain.freeyourgadget.gadgetbridge.service.devices.qhybrid.requests.PlayNotificationRequest;
+
+public class TimePicker extends AlertDialog.Builder {
```

Comment:
A whole custom TimePicker? Why not android.widget.TimePicker? Can you add a Javadoc comment about this if it is really needed?

Reply:
I wanted the user to be able to not select any hour/minute at all, and afaik the default picker doesn't allow that.

Also i wanted to have some specific setting in the same widget...

Reply:
OK, then I suggest adding this reasoning as a javadoc comment. If there's nothing fossil specific in there, I would also move it to a (new) fyg.gadgetbridge.ui package.

Reply:
well, it is quite fossil specific, since only those devices  have the provided vibration patterns...